zebra-state/src/service/non_finalized_state/chain.rs

```diff
@@ -165,15 +167,37 @@ impl UpdateWith<PreparedBlock> for Chain {
             .zip(transaction_hashes.iter().cloned())
             .enumerate()
         {
-            use transaction::Transaction::*;
-            let (inputs, joinsplit_data, sapling_shielded_data) = match transaction.deref() {
+            let (
+                inputs,
+                joinsplit_data,
+                sapling_shielded_data_per_spend_anchor,
+                sapling_shielded_data_shared_anchor,
+                orchard_shielded_data,
+            ) = match transaction.deref() {
                 V4 {
                     inputs,
                     joinsplit_data,
                     sapling_shielded_data,
                     ..
-                } => (inputs, joinsplit_data, sapling_shielded_data),
-                V5 { .. } => unimplemented!("v5 transaction format as specified in ZIP-225"),
+                } => (
+                    inputs,
+                    joinsplit_data,
+                    Some(sapling_shielded_data),
+                    None,
+                    &None,
+                ),
+                V5 {
+                    inputs,
+                    sapling_shielded_data,
+                    orchard_shielded_data,
+                    ..
+                } => (
+                    inputs,
+                    &None,
+                    None,
+                    Some(sapling_shielded_data),
+                    orchard_shielded_data,
+                ),
                 V1 { .. } | V2 { .. } | V3 { .. } => unreachable!(
                     "older transaction versions only exist in finalized blocks pre sapling",
```

Comment:
I think we missed this change when we moved the mandatory checkpoint.

Not really related to this PR, but we should fix it anyway.
```suggestion
                    "older transaction versions only exist in finalized blocks, because of the mandatory canopy checkpoint",
```

Reply:
Was done as part of https://github.com/ZcashFoundation/zebra/pull/2185/commits/39f068db1ae9fd0dccd5cc0ef79214cebff50370 (there were 2 of these).